torchrecipes/paved_path/charnn/main.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env python3
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+import logging
+import os
+import random
+import socket
+import uuid
+from typing import Optional, Tuple
+
+import hydra
+import torch
+import torch.distributed as dist
+from omegaconf import DictConfig
+from torch.nn.parallel import DistributedDataParallel
+from torch.utils.data import random_split
+
+from char_dataset import CharDataset, get_dataset
+from model import GPT, GPTConfig, OptimizerConfig
+from trainer import Checkpoint, Trainer, TrainerConfig, load_checkpoint
+from utils import get_realpath, sample
+
+logger = logging.getLogger(__name__)
+
+
+def get_fq_hostname() -> str:
+    return socket.getfqdn(socket.gethostname())
+
+
+def set_env() -> None:
+    os.environ["RANK"] = os.environ.get("RANK", "0")
+    os.environ["WORLD_SIZE"] = os.environ.get("WORLD_SIZE", "1")
+    os.environ["MASTER_PORT"] = os.environ.get("MASTER_PORT", "29830")
+    os.environ["MASTER_ADDR"] = os.environ.get("MASTER_ADDR", "localhost")
+    os.environ["LOCAL_RANK"] = os.environ.get("LOCAL_RANK", "0")
+    os.environ["TORCHELASTIC_RUN_ID"] = os.environ.get(
+        "TORCHELASTIC_RUN_ID", str(uuid.uuid4()).split("-")[0]
+    )
+
+
+def get_job_name() -> str:
+    uid = os.environ["TORCHELASTIC_RUN_ID"]
+    return f"test-job-{uid}"
+
+
+def get_device() -> torch.device:
+    if  torch.cuda.is_available():
+        local_rank = int(os.environ.get("LOCAL_RANK", "0"))
+        device = torch.device(f"cuda:{local_rank}")
+        torch.cuda.set_device(device)
+    else:
+        device = torch.device("cpu")
+    return device
```

Comment:
should we use the util here? 

https://github.com/pytorch/tnt/blob/5fd77ae9ad8357079cebd14532cb40a2ce8e5761/torchtnt/utils/device.py#L20-L48